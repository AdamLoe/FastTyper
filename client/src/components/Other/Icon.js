import React from "react";
import PT from "prop-types";

let icons = {
  icon1: ({ primary, secondary }) => (
    <>
      <rect
        x="0"
        y="0"
        width="100"
        height="100"
        fill={primary}
        stroke={primary}
      />
      <rect
        x="101"
        y="101"
        width="100"
        height="100"
        fill={primary}
        stroke={primary}
      />
    </>
  ),
  icon2: ({ primary, secondary }) => (
    <>
      <rect
        x="0"
        y="0"
        width="85"
        height="85"
        fill={primary}
        stroke={primary}
      />
      <rect
        x="116"
        y="116"
        width="85"
        height="85"
        fill={primary}
        stroke={primary}
      />
      <rect
        x="0"
        y="116"
        width="85"
        height="85"
        fill={primary}
        stroke={primary}
      />
      <rect
        x="116"
        y="0"
        width="85"
        height="85"
        fill={primary}
        stroke={primary}
      />
    </>
  ),
  icon3: ({ primary, secondary }) => (
    <>
      <circle cx="100" cy="100" r="65" fill={primary} stroke={primary} />
      <circle cx="100" cy="100" r="25" fill={secondary} stroke={secondary} />
    </>
  ),
  icon4: ({ primary, secondary }) => (
    <>
      <rect
        x="85"
        y="0"
        width="30"
        height="200"
        fill={primary}
        stroke={primary}
      />
      <circle cx="100" cy="200" r="80" fill={primary} stroke={primary} />
    </>
  ),
  icon5: ({ primary, secondary }) => (
    <>
      <circle cx="140" cy="140" r="25" fill={primary} stroke={primary} />
      <circle cx="60" cy="70" r="25" fill={primary} stroke={primary} />
      <circle cx="140" cy="70" r="25" fill={primary} stroke={primary} />
      <circle cx="40" cy="192" r="25" fill={primary} stroke={primary} />
    </>
  ),
  icon6: ({ primary, secondary }) => (
    <rect
      x="31"
      y="31"
      width="142"
      height="142"
      fill={primary}
      stroke={primary}
    />
  ),
  icon7: ({ primary, secondary }) => (
    <polygon
      points="100 0, 159 181, 5 69, 195 69, 41 181"
      fill={primary}
      stroke={primary}
      strokeWidth={3}
    />
  )
};

export let iconList = Object.keys(icons);
export let colorList = [
  "white",
  "lightgrey",
  "grey",
  "black",
  "blue",
  "purple",
  "deeppink",
  "red",
  "orange",
  "yellow",
  "green"
];
let random = length => Math.floor(Math.random() * length);
export let getRandomIcon = () => ({
  type: iconList[random(iconList.length)],
  primary: colorList[random(colorList.length)],
  secondary: colorList[random(colorList.length)]
});

let Icon = ({ type, primary, secondary, greyed = false }) => {
  if (!(type in icons)) {
    type = "icon1";
  }

  let IconCore = icons[type];
  let className = greyed ? "greyedIcon" : "";
  return (
    <svg viewBox="-10 -10 220 220" className={className}>
      <circle cx="100" cy="100" r="100" fill={secondary} />
      <IconCore primary={primary} secondary={secondary} />
      <circle
        cx="100"
        cy="100"
        r="110"
        fill="none"
        stroke="black"
        strokeWidth="20"
      />
    </svg>
  );
};
Icon.propTypes = {
  type: PT.string.isRequired,
  primary: PT.string.isRequired,
  secondary: PT.string.isRequired,
  greyed: PT.bool
};

export default Icon;
