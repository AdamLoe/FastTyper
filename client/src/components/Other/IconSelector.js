import React from "react";

import Icon, { iconList } from "./Icon";

let IconSelector = ({ update, type, primary, secondary }) => (
	<div className="iconSelector">
		{ iconList.map( iconType => (
			<div
				className={"selectIconBox " + (type===iconType && " selected ") }
				key={iconType}
				onClick={() => update(iconType)}
			>
				<Icon
					type={iconType}
					primary={primary}
					secondary={secondary}
				/>
			</div>
		))}
	</div>
);

export default IconSelector;