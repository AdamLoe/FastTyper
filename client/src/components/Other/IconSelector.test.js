import React from "react";
import { create } from "react-test-renderer";

import IconSelector from "./IconSelector";
import { iconList } from "./Icon";

let type = "icon5";
let primary = "black";
let secondary = "white";
let update = newType => {
  type = newType;
};
let component = create(
  <IconSelector
    update={update}
    type={type}
    primary={primary}
    secondary={secondary}
  />
);
let root = component.root;

describe("IconSelector", () => {
  test("snapshot", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("update Working", () => {
    iconList.map(icon => {
      root.findByProps({ id: icon }).props.onClick();
      expect(type).toBe(icon);
    });
  });
});
