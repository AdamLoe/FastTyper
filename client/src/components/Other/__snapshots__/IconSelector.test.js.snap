// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconSelector snapshot 1`] = `
<div
  className="iconSelector"
>
  <div
    className="selectIconBox false"
    id="icon1"
    onClick={[Function]}
  >
    <svg
      className=""
      viewBox="-10 -10 220 220"
    >
      <circle
        cx="100"
        cy="100"
        fill="white"
        r="100"
      />
      <rect
        fill="black"
        height="100"
        stroke="black"
        width="100"
        x="0"
        y="0"
      />
      <rect
        fill="black"
        height="100"
        stroke="black"
        width="100"
        x="101"
        y="101"
      />
      <circle
        cx="100"
        cy="100"
        fill="none"
        r="110"
        stroke="black"
        strokeWidth="20"
      />
    </svg>
  </div>
  <div
    className="selectIconBox false"
    id="icon2"
    onClick={[Function]}
  >
    <svg
      className=""
      viewBox="-10 -10 220 220"
    >
      <circle
        cx="100"
        cy="100"
        fill="white"
        r="100"
      />
      <rect
        fill="black"
        height="85"
        stroke="black"
        width="85"
        x="0"
        y="0"
      />
      <rect
        fill="black"
        height="85"
        stroke="black"
        width="85"
        x="116"
        y="116"
      />
      <rect
        fill="black"
        height="85"
        stroke="black"
        width="85"
        x="0"
        y="116"
      />
      <rect
        fill="black"
        height="85"
        stroke="black"
        width="85"
        x="116"
        y="0"
      />
      <circle
        cx="100"
        cy="100"
        fill="none"
        r="110"
        stroke="black"
        strokeWidth="20"
      />
    </svg>
  </div>
  <div
    className="selectIconBox false"
    id="icon3"
    onClick={[Function]}
  >
    <svg
      className=""
      viewBox="-10 -10 220 220"
    >
      <circle
        cx="100"
        cy="100"
        fill="white"
        r="100"
      />
      <circle
        cx="100"
        cy="100"
        fill="black"
        r="65"
        stroke="black"
      />
      <circle
        cx="100"
        cy="100"
        fill="white"
        r="25"
        stroke="white"
      />
      <circle
        cx="100"
        cy="100"
        fill="none"
        r="110"
        stroke="black"
        strokeWidth="20"
      />
    </svg>
  </div>
  <div
    className="selectIconBox false"
    id="icon4"
    onClick={[Function]}
  >
    <svg
      className=""
      viewBox="-10 -10 220 220"
    >
      <circle
        cx="100"
        cy="100"
        fill="white"
        r="100"
      />
      <rect
        fill="black"
        height="200"
        stroke="black"
        width="30"
        x="85"
        y="0"
      />
      <circle
        cx="100"
        cy="200"
        fill="black"
        r="80"
        stroke="black"
      />
      <circle
        cx="100"
        cy="100"
        fill="none"
        r="110"
        stroke="black"
        strokeWidth="20"
      />
    </svg>
  </div>
  <div
    className="selectIconBox  selected "
    id="icon5"
    onClick={[Function]}
  >
    <svg
      className=""
      viewBox="-10 -10 220 220"
    >
      <circle
        cx="100"
        cy="100"
        fill="white"
        r="100"
      />
      <circle
        cx="140"
        cy="140"
        fill="black"
        r="25"
        stroke="black"
      />
      <circle
        cx="60"
        cy="70"
        fill="black"
        r="25"
        stroke="black"
      />
      <circle
        cx="140"
        cy="70"
        fill="black"
        r="25"
        stroke="black"
      />
      <circle
        cx="40"
        cy="192"
        fill="black"
        r="25"
        stroke="black"
      />
      <circle
        cx="100"
        cy="100"
        fill="none"
        r="110"
        stroke="black"
        strokeWidth="20"
      />
    </svg>
  </div>
  <div
    className="selectIconBox false"
    id="icon6"
    onClick={[Function]}
  >
    <svg
      className=""
      viewBox="-10 -10 220 220"
    >
      <circle
        cx="100"
        cy="100"
        fill="white"
        r="100"
      />
      <rect
        fill="black"
        height="142"
        stroke="black"
        width="142"
        x="31"
        y="31"
      />
      <circle
        cx="100"
        cy="100"
        fill="none"
        r="110"
        stroke="black"
        strokeWidth="20"
      />
    </svg>
  </div>
  <div
    className="selectIconBox false"
    id="icon7"
    onClick={[Function]}
  >
    <svg
      className=""
      viewBox="-10 -10 220 220"
    >
      <circle
        cx="100"
        cy="100"
        fill="white"
        r="100"
      />
      <polygon
        fill="black"
        points="100 0, 159 181, 5 69, 195 69, 41 181"
        stroke="black"
        strokeWidth={3}
      />
      <circle
        cx="100"
        cy="100"
        fill="none"
        r="110"
        stroke="black"
        strokeWidth="20"
      />
    </svg>
  </div>
</div>
`;
